openpathsampling/snapshot.py

```diff
@@ -575,7 +586,103 @@ def subset(self, subset):
+            momentum=self.momentum.copy(subset),
+            is_reversed=self.is_reversed,
+            topology=self.topology.subset(subset)
+        )
         return this
+
+
+@lazy_loading_attributes('_reversed')
+class ToySnapshot(AbstractSnapshot):
+    """
+    Simulation snapshot. Contains references to a configuration and momentum
+    """
+
+    __features__ = ['Velocities', 'Coordinates']
+
+    def __init__(self, coordinates=None, velocities=None, is_reversed=False, topology=None,
+                 reversed_copy=None, engine=None):
+        """
+        Create a simulation snapshot. Initialization happens primarily in
+        one of two ways:
+            1. Specify `Configuration` and `Momentum` objects
+            2. Specify the things which make up `Configuration` and
+               `Momentum` objects, i.e., coordinates, velocities, box
+               vectors, etc.
+        If you want to obtain a snapshot from a currently-running MD engine,
+        use that engine's .current_snapshot property.
+
+        Parameters
+        ----------
+        coordinates : simtk.unit.Quantity wrapping Nx3 np array of dimension length
+            atomic coordinates (default: None)
+        velocities : simtk.unit.Quantity wrapping Nx3 np array of dimension length
+            atomic velocities (default: None)
+        box_vectors : periodic box vectors (default: None)
+            the periodic box vectors at current timestep (defautl: None)
+        potential_energy : simtk.unit.Quantity of units energy/mole
+            potential energy at current timestep (default: None)
+        kinetic_energy : simtk.unit.Quantity of units energy/mole
+            kinetic energy at current timestep (default: None)
+
+        Attributes
+        ----------
+        coordinates : simtk.unit.Quantity wrapping Nx3 np array of dimension length
+            atomic coordinates
+        velocities : simtk.unit.Quantity wrapping Nx3 np array of dimension length
+            atomic velocities
+        box_vectors : periodic box vectors
+            the periodic box vectors
+        potential_energy : simtk.unit.Quantity of units energy/mole
+            potential energy
+        kinetic_energy : simtk.unit.Quantity of units energy/mole
+            kinetic energy
+        idx : dict( Storage() : int )
+            dict for storing the used index per storage
+        """
+
+        super(ToySnapshot, self).__init__(is_reversed, reversed_copy, topology)
+
+        self.coordinates = coordinates
+        self.velocities = velocities
+
+        if reversed_copy is None:
+            self._reversed.coordinates = self.coordinates
+            self._reversed.velocities = -1.0 * self.velocities
+
+
+    # TODO: Can be removed. We don't need this anymore. Still here for legacy reasons
+    @property
+    def xyz(self):
+        """
+        Coordinates without dimensions.
+
+        """
+        return self.coordinates
```

Comment:
The toy engine exists as a simple model that should be analogous to MD. Therefore, the `ToySnapshot` should implement as much of the same API as `Snapshot` as possible. Keep .xyz around (although it can be a one-line alias `xyz = coordinates` now).